### PR TITLE
Space-Travelers-Hub Project 

### PR DESCRIPTION
**Pull request by rahimialijan**
# Space-Travelers-Hub Project 

- Cleaned linter errors
- Add all required components
- Navigation is working properly
- Add test case for components
- use react and redux to develop the App

### CODE REVIEW

File: src/components/Missions/missionsDisplay.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        <tbody>
+          {missions.map((mission, index) => (
+            <tr
+              key={mission.mission_id}
+              className={index % 2 === 0 ? 'even-row' : 'odd-row'}
+            >
+              <td className="bold">{mission.mission_name}</td>
+              <td>{mission.description}</td>
+              <td className="button-cell">
+                {mission.reserved ? (
```
Comment (t-yanick) [inline src/components/Missions/missionsDisplay.js:45]: - [ ] As per project requirements you are to use React conditional rendering syntax for Missions and Rockets. At the moment you are using it only for Rockets. Kindly extent its usage to Missions.
image

File: src/router/rockets.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { useSelector } from 'react-redux';
+import RocketItem from '../components/Rockets/rockets';
+
+const Rockets = () => {
+  const { rockets, isLoading, error } = useSelector((store) => store.rockets);
+  // const dispatch = useDispatch();
+  // useEffect(() => {
+  //   dispatch(getRocketsData());
+  // }, [dispatch]);
```
Comment (t-yanick) [inline src/router/rockets.js:9]: - [ ] Kindly remove all commented pieces of code from your codebase

File: src/router/rockets.test.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+jest.mock('axios');
+
+it('renders Correctly', () => {
+  const tree = render(
+    <Provider store={store}>
+      <Rockets />
+    </Provider>,
+  );
+  expect(tree).toMatchSnapshot();
+});
```
Comment (t-yanick) [inline src/router/rockets.test.js:16]: - [ ] Great job with testing. Kindly add a snap also for missions. `...../missions.test.js`

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @AleWaweru and @rahimialijan , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Collaborative work :+1: 
:heavy_check_mark: App works as expected
:heavy_check_mark: Proper routes and view components should be created
:heavy_check_mark: `useSelector()` should used to get a state slice corresponding to routes
:heavy_check_mark: Use of `filter()` and `map()` to update the state

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 4]
```diff
@@ -1,70 +1,173 @@
...
 
-**Note: this is a one-way operation. Once you `eject`, you can't go back!**
+- **[Initialize React app]**
+- **[React Components]**
 
-If you aren't satisfied with the build tool and configuration choices, you can `eject` at any time. This command will remove the single build dependency from your project.
+### Live Demo <a name="live-demo"></a>
 
-Instead, it will copy all the configuration files and the transitive dependencies (webpack, Babel, ESLint, etc) right into your project so you have full control over them. All of the commands except `eject` will still work, but they will point to the copied scripts so you can tweak them. At this point you're on your own.
+- **[Live Demo](https://rahimialijan.github.io/Space-Travelers-Hub/)**
```
[the remark this answers is not in the document]
Comment (AleWaweru) [reply README.md:59]: done.

[the remark this answers is not in the document]
Comment (AleWaweru) [reply README.md:59]: done

File: src/components/Rockets/RocketsProfile/rocketProfile.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import { useSelector } from 'react-redux';
+import './rocketProfile.css';
+
+const RocketProfile = () => {
+  const { reservedRockets } = useSelector((state) => state.rockets);
+  // const dispatch = useDispatch();
+  // useEffect(() => {
+  //   dispatch(getRocketsData());
+  // }, [dispatch]);
```
[the remark this answers is not in the document]
Comment (AleWaweru) [reply src/components/Rockets/RocketsProfile/rocketProfile.js:9]: done

File: package.json [hunk 6]
```diff
@@ -1,17 +1,26 @@
+
 {
   "name": "spacetravelershub",
   "version": "0.1.0",
   "private": true,
   "dependencies": {
```
[the remark this answers is not in the document]
Comment (AleWaweru) [reply package.json:6]: done

File: src/components/Missions/missionsDisplay.js (hunk 1 above)
Comment (AleWaweru) [reply src/components/Missions/missionsDisplay.js:45]: done

File: src/router/rockets.js (hunk 2 above)
Comment (AleWaweru) [reply src/router/rockets.js:9]: done

Comment (AleWaweru) [reply src/router/rockets.js:9]: done

File: src/router/rockets.test.js (hunk 3 above)
Comment (AleWaweru) [reply src/router/rockets.test.js:16]: done.

File: src/components/Missions/missionsDisplay.js (hunk 1 above)
Comment (AleWaweru) [reply src/components/Missions/missionsDisplay.js:45]: I noticed  that my code actually used react conditional rendering syntax

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**